app/main.py

```diff
@@ -1,2 +1,72 @@
+from __future__ import annotations
+from typing import Hashable, Any
+
+
+class Node:
+    def __init__(self, key: Hashable, hash_val: int, value: Any) -> None:
+        self.key = key
+        self.hash_val = hash_val
+        self.value = value
+
+
+def _is_matching_key(node: Node, key: Hashable) -> bool:
+    return node.key == key and node.hash_val == hash(key)
+
+
 class Dictionary:
-    pass
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self.size = 0
+        self.capacity: int = 8
+        self.table: list = [None] * self.capacity
+
+    def get_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+        while (
+                self.table[index] and not
+                _is_matching_key(self.table[index], key)):
+            index = (index + 1) % self.capacity
+        return index
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size >= self.capacity * self.LOAD_FACTOR:
+            self.resize()
+        index = self.get_index(key)
+        while self.capacity <= index:
+            self.resize()
```

Comment:
Why you need resize in this case?????

Reply:
If the index exceeds the current capacity, it means that the table is full or close to full, and inserting a new element may result in poor performance or inefficient storage. So, it triggers a resize operation to increase the capacity of the table, allowing it to accommodate more elements efficiently.

Reply:
You are resizing in line 34, so there will not be case, when Index would be out of range. Also, it's better to perform a resize after you added element, mean in the end of this function, because size increases there.